mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java

```diff
@@ -1,13 +1,259 @@
+
+            final Resources res = itemView.getContext().getResources();
+            mRowHeightPixels = res.getDimensionPixelSize(R.dimen.row_height);
+        }
+
+        @Override
+        public void bind(Widget widget) {
+            super.bind(widget);
+
+            ViewGroup.LayoutParams lp = mMapView.getLayoutParams();
+            int rows = widget.height() > 0 ? widget.height() : 5;
+            int desiredHeightPixels = rows * mRowHeightPixels;
+            if (lp.height != desiredHeightPixels) {
+                lp.height = desiredHeightPixels;
+                mMapView.setLayoutParams(lp);
+            }
+
+            mBoundItem = widget.item();
+            applyPositionAndLabelWhenReady(mMapView, 15.0f, false, false);
+        }
+
+        @Override
+        public void start() {
+            super.start();
+            if (!mStarted) {
+                mMapView.onResume();
+                mStarted = true;
+            }
+        }
+
+        @Override
+        public void stop() {
+            super.stop();
+            if (mStarted) {
+                mMapView.onPause();
+                mStarted = false;
+            }
+        }
+
+        @Override
+        public void onMarkerDragStart(Marker marker) {
+            // no-op, we're interested in drag end only
+        }
+
+        @Override
+        public void onMarkerDrag(Marker marker) {
+            // no-op, we're interested in drag end only
+        }
+
+        @Override
+        public void onMarkerDragEnd(Marker marker) {
+            String newState = String.format(Locale.US, "%f,%f",
+                    marker.getPosition().getLatitude(), marker.getPosition().getLongitude());
+            String item = marker.getId();
+            Util.sendItemCommand(mConnection.getAsyncHttpClient(), item, newState);
+        }
+
+        private void openPopup() {
+            final MapView mapView = new MapView(itemView.getContext());
+
+            AlertDialog dialog = new AlertDialog.Builder(itemView.getContext())
+                    .setView(mapView)
+                    .setCancelable(true)
+                    .setNegativeButton(R.string.close, null)
+                    .create();
+
+            dialog.setOnDismissListener(dialogInterface -> mapView.onPause());
+            dialog.setCanceledOnTouchOutside(true);
+            dialog.show();
+
+            mapView.setBuiltInZoomControls(true);
+            mapView.setMultiTouchControls(true);
+            mapView.setVerticalMapRepetitionEnabled(false);
+            mapView.getOverlays().add(new CopyrightOverlay(itemView.getContext()));
+            mapView.onResume();
+            applyPositionAndLabelWhenReady(mapView, 16.0f, true, true);
+        }
+
+        private void applyPositionAndLabelWhenReady(MapView mapView, float zoomLevel,
+                boolean allowDrag, boolean allowScroll) {
+            mHandler.post(() -> applyPositionAndLabel(mapView, zoomLevel, allowDrag, allowScroll));
```

Comment:
Still not too happy with this, but I realize it's an upstream issue. Should probably merge this now and revisit when they fix this.